src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1604,13 +1577,13 @@ private bool ExactConstructedInference(TypeSymbol source, TypeSymbol target, ref
             // SPEC:   type C<U1...Uk> then an exact inference 
             // SPEC:   is made from each Ui to the corresponding Vi.
 
-            var namedSource = source as NamedTypeSymbol;
+            var namedSource = source.TupleUnderlyingTypeOrSelf() as NamedTypeSymbol;
```

Comment:
NIT: I'd put this in the caller to be sure unwrap is not called when we do not expect. However, since there is only one caller, unwrapping here seems ok as well.